src/EpisodeIntegrationService/ProcessData/ProcessData.cs

```diff
@@ -125,64 +153,44 @@ private async Task ProcessEpisodeDataAsync(List<BssEpisode> episodes, string epi
 
                 string serializedEpisode = JsonSerializer.Serialize(modifiedEpisode, new JsonSerializerOptions { WriteIndented = true });
 
-                // Log the Episode data before sending it
                 _logger.LogInformation($"Sending Episode to {episodeUrl}: {serializedEpisode}");
 
                 await _httpRequestService.SendPost(episodeUrl, serializedEpisode);
             }
         }
-        else
+        catch (Exception ex)
         {
-            _logger.LogInformation("No episode data found.");
+            _logger.LogError("Error in ProcessEpisodeDataAsync: " + ex.Message);
+            ProcessEpisodeDataAsync(episodes, episodeUrl);
```

Comment:
I think the try catch needs to be moved inside the foreach. That way when it catches an exception it can log it but then continue with the next episode. Currently if an exception keeps occurring for a particular episode it's going to get stuck in a loop

Reply:
Discussed with Alex in a call